modules/core/src/test/java/org/apache/ignite/internal/processors/security/cache/ContinuousQueryPermissionCheckTest.java

```diff
@@ -0,0 +1,133 @@
+import org.apache.ignite.plugin.security.SecurityException;
+import org.apache.ignite.plugin.security.SecurityPermissionSet;
+import org.apache.ignite.plugin.security.SecurityPermissionSetBuilder;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import static org.apache.ignite.plugin.security.SecurityPermission.CACHE_READ;
+import static org.apache.ignite.testframework.GridTestUtils.assertThrowsWithCause;
+
+/**
+ * Test cache permission for Continuous Queries.
+ */
+@RunWith(JUnit4.class)
+public class ContinuousQueryPermissionCheckTest extends AbstractCacheOperationPermissionCheckTest {
+    /** Test server node name. */
+    public static final String SRV = "srv_test_node";
+
+    /** Test client node name. */
+    public static final String CLNT = "cnt_test_node";
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        Ignite srv = startGridAllowAll("server");
+
+        startGrid(SRV, permissions(), true);
+
+        startGrid(CLNT, permissions(), true);
+
+        srv.cluster().active(true);
+    }
+
+    /** Test ContinuousQuery from a client. */
+    @Test
+    public void testClientContinuousQuery() {
+        check(n -> startContinuousQuery(grid(CLNT), n));
+    }
+
+    /** Test ContinuousQuery from a server. */
+    @Test
+    public void testServerContinuousQuery() {
+        check(n -> startContinuousQuery(grid(SRV), n));
+    }
+
+    /** Test ContinuousQueryWithTransformer from a client. */
+    @Test
+    public void testClientContinuousQueryWithTransformer() {
+        check(n -> startContinuousQueryWithTransformer(grid(CLNT), n));
+    }
+
+    /** Test ContinuousQueryWithTransformer from a server. */
+    @Test
+    public void testServerContinuousQueryWithTransformer() {
+        check(n -> startContinuousQueryWithTransformer(grid(SRV), n));
+    }
+
+    /**
+     * @param c Consumer that accepts a cache name.
+     */
+    private void check(final Consumer<String> c) {
+        c.accept(CACHE_NAME);
+
+        assertThrowsWithCause(() -> c.accept(FORBIDDEN_CACHE), SecurityException.class);
+    }
+
+    /** */
+    private void startContinuousQuery(Ignite node, String cacheName) {
+        ContinuousQuery<String, Integer> q = new ContinuousQuery<>();
+
+        q.setLocalListener(e -> {/* No-op. */});
+
+        try (QueryCursor<Cache.Entry<String, Integer>> cur = node.cache(cacheName).query(q)) {
+            // No-op.
+        }
+    }
+
+    /** */
+    private void startContinuousQueryWithTransformer(Ignite node, String cacheName) {
+        ContinuousQueryWithTransformer<String, Integer, String> q = new ContinuousQueryWithTransformer<>();
+
+        q.setLocalListener(e -> {/* No-op. */});
+
+        q.setRemoteTransformerFactory(new TestTransformerFactory());
+
+        try (QueryCursor<Cache.Entry<String, Integer>> cur = node.cache(cacheName).query(q)) {
+            // No-op.
+        }
+    }
+
+    /** Permissions */
```

Comment:
Convert to multi-line javadoc according to coding guidelines, + '.' at the EOL, + @return tag.